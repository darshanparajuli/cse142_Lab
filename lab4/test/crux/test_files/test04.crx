// test index operator

array x: int[10][10][10];

func main() : void {
  ::printInt(x[2][3][4]);
}
