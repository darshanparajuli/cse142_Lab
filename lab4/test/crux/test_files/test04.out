ast.DeclarationList(3,1)
  ast.ArrayDeclaration(3,1)[Symbol(x)]
  ast.FunctionDefinition(5,1)[Symbol(main), []]
    ast.StatementList(6,3)
      ast.Call(6,3)[Symbol(printInt)]
        ast.ExpressionList(6,14)
          ast.Dereference(6,14)
            ast.Index(6,22)
              ast.Index(6,19)
                ast.Index(6,16)
                  ast.AddressOf(6,14)[x]
                  ast.LiteralInt(6,16)[2]
                ast.LiteralInt(6,19)[3]
              ast.LiteralInt(6,22)[4]

